README.md

```diff
@@ -390,43 +402,55 @@ public interface ExampleHandler {
 
 ##### Custom-named
 
-It is a sort of combination of `Named` and `Indexed` annotations. You can get needed value by index or by key. Can be multi-level
+It is a sort of combination of `Named` and `Indexed` annotations. You can get
+needed value by index or by key. It also allows getting elements from nested
+objects and arrays, the value will be got in the order the keys and indexes
+are specified. To get a value by key, you should just declare the required key
+like in `@Named` annotation. To get an array by index, you can declare it as
+`[index]`, in such a way you can get values from this array like `{inner index}`.
 
 ```java
 @JSTPHandler
 public interface ExampleHandler {
   // ...
 
   @CustomNamed("ok")
-  void onNeededValue(JSValue args);
+  void onNeededNamedValue(JSValue args);
 
   @CustomNamed("{1}")
-  void onNeededValue(JSValue args);
+  void onNeededIndexValue(JSValue args);
 
   // gets packet["ok"][1][2]
   @CustomNamed(value = {"ok", "[1]", "{2}"})
-  void onNeededValue(JSValue args);
+  void onNeededMixValue(JSValue args);
   // ...
 }
 ```
 
-After compilation class `JSTPExampleHandler` will be generated and you will be able to use it in packets processing.
+After compilation class named like `JSTP + (YourHandlerName)` (for this example
+it will be `JSTPExampleHandler`) will be generated and you will be able to use
+it in packet processing.
 
 ```java
 connection.call("interfaceName", "methodName", args, new JSTPExampleHandler() {
     // ...
 });
 ```
 #### JSTP receiver
-You can process received values not only via single handler, but by several ones. They can be added or removed to certain `JSTP receiver`, generated similarly to `JSTPHandler`. To generate receiver, you need to do the following:
+You can process received values not only via single handler, but by several
+ones. They can be added or removed to certain `JSTP receiver`, generated
```

Comment:
to certain `JSTP receiver` -> from `JSTP receiver` via \<proper method names\>.
And I think 'generated..' should be separate sentence as it is not related to this one.